### PR TITLE
chore: prettier with types and format the entire codebase

### PR DESCRIPTION
**Pull request by ahkhanjani**
### Reasons for making this change

- Prettier config file now uses types.
- All the previous Prettier rules that contained the default value (5 out of 5 rules) have been removed.
- Added single quote rule for JS and JSX.
- Ran Prettier even outside the package scope.

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [ ] **I'm adding or updating code**
  - [ ] I've added and/or updated tests. I've run `npm run test:update` to update snapshots, if needed.
  - [ ] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
  - [ ] I've updated the [changelog](https://github.com/rjsf-team/react-jsonschema-form/blob/main/CHANGELOG.md) with a description of the PR
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

File: prettier.config.cjs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/** @type {import("prettier").Config} */
```
Comment (heath-freenome) [inline prettier.config.cjs:1]: Why cjs? could you use typescript or javascript instead?

**Comment by heath-freenome** [comment]
@ahkhanjani I have questions:

1. You removed husky which enable the linting/formatting precommit hook for new code, why?
2. I don't see how you prettied up the code without the precommit hook, did you run it manually?
3. If yes to 2, we want future code to fit the styling (hence the precommit hook), how do you propose we do it?

File: prettier.config.cjs (hunk 1 above)
Comment (ahkhanjani) [reply prettier.config.cjs:1]: > Why cjs? could you use typescript or javascript instead?

They simply don't support it. From Prettier docs:

> - A "prettier" key in your package.json file.
> - A .prettierrc file written in JSON or YAML.
> - A .prettierrc.json, .prettierrc.yml, .prettierrc.yaml, or .prettierrc.json5 file.
> - A .prettierrc.js, .prettierrc.cjs, prettier.config.js, or prettier.config.cjs file that exports an object using module.exports.
> - A .prettierrc.toml file.

cjs is currently the closest we have to ts

**Comment by heath-freenome** [comment]
Also, there are two PRs ahead of this that you will have to rebase into your MR and redo the formatting I believe

**Comment by ahkhanjani** [comment]
@heath-freenome In long term we would want to get rid of precommit hooks and rely on CI for that. Precommit hooks are painful for local testing flow. I'd like to know your opinion on this matter

> 2\. I don't see how you prettied up the code without the precommit hook, did you run it manually?

Yes. I ran `prettier --write .` because husky didn't format anything outside the packages folder.

> 3\. If yes to 2, we want future code to fit the styling (hence the precommit hook), how do you propose we do it?

At the moment I'm not sure what the options are without the precommit hooks. I'll let you know if I figure it out

**Comment by heath-freenome** [comment]
> @heath-freenome In the long term we would want to get rid of precommit hooks and rely on CI for that. Precommit hooks are painful for local testing flow. I'd like to know your opinion on this matter
Personally I like the precommit hook since it prevents CI from failing with lint errors since it was pre-checked. I've seen people pushing code that fails CI because they didn't properly install the precommit hooks. Sometimes they never even bother to reformat their code because they seem to have the "Once I've pushed it I'm done" attitude. 

> > 3. If yes to 2, we want future code to fit the styling (hence the precommit hook), how do you propose we do it?
> 
> At the moment I'm not sure what the options are without the precommit hooks. I'll let you know if I figure it out
I'm pretty sure that formatting code and committing code entirely with CI isn't doable. So it really is up to the person pushing a PR to ensure everything is formatted. But maybe things have improved on Github to allow this?

**Comment by heath-freenome** [comment]
@ahkhanjani Ok, all PRs are merged. Feel free to rebase this one now.

**Comment by ahkhanjani** [comment]
I'll make another PR as I messed some things up with this one

**Pull request closed without merging**